### PR TITLE
Enable Java driver release for 3.0

### PR DESCRIPTION
**Pull request by farost**
## Usage and product changes
Uncommented JNI and Java Driver jobs for the `circleci` config.
Updated release notes for the 3.0 release.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -700,39 +700,39 @@ jobs:
       - checkout
       - install-bazel-yum:
           bazel-arch: arm64
-      - deploy-pip-release-unix
+#      - deploy-pip-release-unix
       - deploy-maven-jni-release-unix
-      - deploy-clib-release-unix
```
Comment (farost) [inline .circleci/config.yml:705]: It shouldn't be needed for the JNI work as everything is a part of the JNI library, right? Is it just for the standalone C library?

Comment (farost) [reply .circleci/config.yml:705]: (the C driver is not ready, I only touched the path to the outer drivers)

Comment (flyingsilverfin) [reply .circleci/config.yml:705]: Yes it doesn't have to be released for JNI to work, we just need the JNI to bundle the c library

**Review by flyingsilverfin — APPROVED** [review]

File: RELEASE_NOTES_LATEST.md [hunk 2]
```diff
@@ -29,66 +29,28 @@ Documentation: https://typedb.com/docs/drivers/java/overview
     <dependency>
         <groupid>com.vaticle.typedb</groupid>
```
Comment (krishnangovindraj) [inline RELEASE_NOTES_LATEST.md:30]: This is out of scope for today, but we should decide if we want to shrink this from
`com.vaticle.typedb` -> `com.typedb`

**Review by krishnangovindraj — APPROVED** [review]
looks good to me

File: RELEASE_NOTES_LATEST.md (hunk 2 above)
Comment (farost) [reply RELEASE_NOTES_LATEST.md:30]: You're right, I'll at least add a task for it!

**Pull request merged**